### PR TITLE
Features/add tabs and tab component 214

### PR DESCRIPTION
**Pull request by nfiniteset**
tabs-orion

## Description
- Add Tabs and Tab
- Enable Adding Tabs as a child of SubNav
- Add buttons to manipulate tabs in the playground

## Motivation and Context
Implement a HIG.web component

## How Has This Been Tested?
Not yet...

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: packages/react-hig/src/elements/components/GlobalNav/Tab.js [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+  }
+
+  commitUpdate(updatePayload, oldProps, newProp) {
+    for (let i = 0; i < updatePayload.length; i += 2) {
+      const propKey = updatePayload[i];
+      const propValue = updatePayload[i + 1];
+
+      switch (propKey) {
+        case 'label':
+          this.hig.setTabel(propValue);
```
Comment (unitedempire) [inline packages/react-hig/src/elements/components/GlobalNav/Tab.js]: This should be ```this.hig.setLabel(propValue);``` ?

Comment (nfiniteset) [inline packages/react-hig/src/elements/components/GlobalNav/Tab.js]: Yep.

**Comment by camwest** [comment]
Feedback:

* Since Tabs is a singleton child, add a check for a case where there is multiple Tabs components as children of SubNav (see GlobaNav#appendChild for a reference).
* removeChild should also clean up the this.tabs field (see GlobalNav#removeChild)
* For Tab -> Check the HIGElement base class for some helpful methods (like commitPropChange) which makes it easier to set up change commits. Look at https://github.com/Autodesk/orion-ui/blob/master/packages/react-hig/src/elements/basics/Button.js for a more up-to-date example
* Great job adding the example to the index!

**Comment by nfiniteset** [comment]
I've addressed your comments above, added unit tests for Tabs, Tab, and SubNav.

File: packages/react-hig/src/elements/components/GlobalNav/Tab.js [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+          break;
+        default: {
+          this.commitPropChange(propKey, propValue);
+        }
+      }
+    }
+  }
+}
+
+const TabComponent = createComponent(Tab);
```
Comment (camwest) [inline packages/react-hig/src/elements/components/GlobalNav/Tab.js]: Can you add propTypes and __docgenInfo for the TabComponent?

File: packages/react-hig/src/elements/components/GlobalNav/Tabs.js [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+        /* no-op */
+        break;
+      default:
+        throw new Error(`Unknown type ${ElementConstructor.name}`);
+    }
+  }
+
+  appendChild(instance) {
+    if (instance instanceof Tab) {
+      if (this.mounted) {
```
Comment (camwest) [inline packages/react-hig/src/elements/components/GlobalNav/Tabs.js]: You should ALWAYS be appending to the HIGNodeList. You're only appending here when it's mounting which will result in an inconsistent model when using insertBefore.

Comment (camwest) [inline packages/react-hig/src/elements/components/GlobalNav/Tabs.js]: See https://github.com/Autodesk/orion-ui/blob/f75e581e1ed562f3aba7837b02f2400c1c90421e/packages/react-hig/src/elements/components/GlobalNav/SectionList.js#L57 for a reference

File: packages/react-hig/src/elements/components/GlobalNav/Tabs.js [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+
+  insertBefore(instance, insertBeforeIndex) {
+    const beforeChild = this.tabs.item(insertBeforeIndex);
+    this.tabs.insertBefore(instance, beforeChild);
+    this.hig.addTab(instance.hig, beforeChild.hig);
+    instance.mount();
+  }
+
+  removeChild(instance) {
+    instance.unmount();
```
Comment (camwest) [inline packages/react-hig/src/elements/components/GlobalNav/Tabs.js]: You should be removing the instance from the HIGNodeList here or else it'll give an inconsistent model when using insertBefore

File: packages/react-hig/src/elements/components/GlobalNav/Tabs.js [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+  }
+}
+
+const TabsComponent = createComponent(Tabs);
```
Comment (camwest) [inline packages/react-hig/src/elements/components/GlobalNav/Tabs.js]: Add propTypes and __docgenInfo

**Comment by camwest** [comment]
Related Feature #214

**Comment by nfiniteset** [comment]
Ok @camwest I've got docgenInfo and propTypes in place for Tabs and Tab. Will merge if the build is green.

**Pull request merged**